### PR TITLE
Include options in apply_sort

### PR DESCRIPTION
**Pull request by johantell**
Including options will allow to create sorting based on the actual context. It's especially useful when working with scopes that need request based parameters. In my case position or the current_user

### CODE REVIEW

**Comment by xdmx** [comment]
:+1:

File: lib/jsonapi/resource.rb [hunk 1]
```diff
@@ -486,7 +486,7 @@ def apply_pagination(records, paginator, order_options)
         records
       end
 
-      def apply_sort(records, order_options)
+      def apply_sort(records, order_options, options = {})
```
Comment (lgebhardt) [inline lib/jsonapi/resource.rb]: Could you rename `options` to `_options` for style consistency as it's unused in this method?

Comment (johantell) [inline lib/jsonapi/resource.rb]: certainly!

**Comment by lgebhardt** [comment]
One more change discovered in #561. We should pass the context in as an option on line 864 (https://github.com/cerebris/jsonapi-resources/blob/master/lib/jsonapi/resource.rb#L864) so it's available when we are sorting on related records.

``` ruby
records = resource_klass.apply_sort(records, order_options, context: @context)
```

**Comment by johantell** [comment]
Should we consider adding some information to the readme?

**Comment by lgebhardt** [comment]
@mintcore I think adding some information to the readme would be good. I also wonder if we should pass the options provided to relationship_name, but with the context overridden. Or maybe just make `apply_sort` take context instead of options. I think I may like this better as the set of options won't be consistent between the two calls. Thoughts?

**Comment by johantell** [comment]
I agree about the context part!

File: lib/jsonapi/resource.rb [hunk 2]
```diff
@@ -544,7 +544,7 @@ def find(filters, options = {})
 
         sort_criteria = options.fetch(:sort_criteria) { [] }
         order_options = construct_order_options(sort_criteria)
-        records = sort_records(records, order_options)
+        records = sort_records(records, order_options, options)
```
Comment (lgebhardt) [inline lib/jsonapi/resource.rb]: `options` => `context`

File: test/unit/resource/resource_test.rb [hunk 3]
```diff
@@ -240,7 +240,7 @@ def test_to_many_relationship_sorts
 
     # define apply_filters method on post resource to not respect filters
     PostResource.instance_eval do
-      def apply_sort(records, criteria)
+      def apply_sort(records, criteria, options = {})
```
Comment (lgebhardt) [inline test/unit/resource/resource_test.rb]: `options` => `context`

File: test/unit/resource/resource_test.rb [hunk 4]
```diff
@@ -253,7 +253,7 @@ def apply_sort(records, criteria)
   ensure
     # reset method to original implementation
     PostResource.instance_eval do
-      def apply_sort(records, criteria)
+      def apply_sort(records, criteria, options = {})
```
Comment (lgebhardt) [inline test/unit/resource/resource_test.rb]: `options` => `context`

**Comment by lgebhardt** [comment]
@mintcore Thanks!

**Pull request merged**